ai-engineering/measure.mdx

```diff
@@ -1,85 +1,374 @@
+        content: `You are a customer support engineer classifying tickets as: spam, question, feature_request, or bug_report.
+        
+If spam, return a polite auto-close message. Otherwise, say a team member will respond shortly.`,
       },
       {
-        input: 'foobar',
-        expected: 'hello, foobar!',
+        role: 'user',
+        content: subject ? `Subject: ${subject}\n\n${content}` : content,
       },
-    ];
-  },
+    ],
+    schema: ticketResponseSchema,
+  });
 
-  // 2. The task that runs your capability
-  task: async (input: string) => {
-    return `hi, ${input}!`;
+  return result.object;
+}
+
+// Custom exact-match scorer
+const ExactMatchScorer = Scorer(
+  'Exact-Match',
+  ({ output, expected }: { output: { response: string }; expected: { response: string } }) => {
+    return ExactMatch({
+      output: output.response,
+      expected: expected.response,
+    });
+  }
+);
+
+// Custom spam classification scorer
+const SpamClassificationScorer = Scorer(
+  'Spam-Classification',
+  ({ output, expected }: { 
+    output: { category: string }; 
+    expected: { category: string };
+  }) => {
+    return (expected.category === 'spam') === (output.category === 'spam') ? 1 : 0;
+  }
+);
+
+// Define the evaluation
+Eval('spam-classification', {
+  // Specify which flags this eval uses
+  configFlags: pickFlags('ticketClassification'),
+  
+  // Test data with input/expected pairs
+  data: () => [
+    {
+      input: {
+        subject: "Congratulations! You've Been Selected for an Exclusive Reward",
+        content: 'Claim your $500 gift card now by clicking this link!',
+      },
+      expected: {
+        category: 'spam',
+        response: "We're sorry, but your message has been automatically closed.",
+      },
+    },
+    {
+      input: {
+        subject: 'FREE CA$H',
+        content: 'BUY NOW ON WWW.BEST-DEALS.COM!',
+      },
+      expected: {
+        category: 'spam',
+        response: "We're sorry, but your message has been automatically closed.",
+      },
+    },
+  ],
+  
+  // The task to run for each test case
+  task: async ({ input }) => {
+    return await classifyTicket(input);
+  },
+  
+  // Scorers to measure performance
+  scorers: [SpamClassificationScorer, ExactMatchScorer],
+  
+  // Optional metadata
+  metadata: {
+    description: 'Classify support tickets as spam or not spam',
   },
+});
+```
+
+## Set up scorers
+
+A scorer is a function that scores a capability’s output. Scorers receive the `input`, the generated `output`, and the `expected` value, and return a score (typically 0-1).
+
+Examples to set up scorers:
+- Use a simple exact match scorer that returns 1 if the output matches the expected value, and 0 otherwise:
+
+    ```ts
+    import { Scorer } from 'axiom/ai/evals';
+
+    const ExactMatchScorer = Scorer(
+      'Exact-Match',
+      ({ output, expected }: { output: string; expected: string }) => {
+        return output === expected ? 1 : 0;
+      }
+    );
```

Comment:
this (or something like it) is what we should be doing in the evals quickstart. showing autoevals there (which is a braintrust product!) makes the code more opaque for no good reason.